invokeai/frontend/web/src/features/nodes/store/util/parseFieldType.ts

```diff
@@ -0,0 +1,14 @@
+import { FieldType } from 'features/nodes/types/types';
+
+export const getIsPolymorphic = (type: FieldType | string): boolean =>
+  type.endsWith('Polymorphic');
+
+export const getIsCollection = (type: FieldType | string): boolean =>
+  type.endsWith('Collection');
+
+export const getBaseType = (type: FieldType | string): FieldType | string =>
+  getIsPolymorphic(type)
+    ? type.replace(/Polymorphic$/, '')
+    : getIsCollection(type)
+    ? type.replace(/Collection$/, '')
+    : type;
```

Comment:
I'm certain that I'm not aware of all of the constraints here, but seeing this makes me think that we should definitely try to move toward a structured type representation rather than relying on type names for everything.

i.e. something like:
```typescript
type FieldType = {
  name: string;
  isCollection: boolean;
  isPolymorphic: boolean;
  // Maybe?
  isCustom: boolean;
}
```

Is this something you're already thinking about? Are there constraints that make this hard to do today?

Reply:
Yeah I was wishing I made it like that initially. The collection and polymorphic support wasn't in the initial design so it was tacked on like you see now.

The only constraint is that existing workflows have field types baked in to them - another thing that probably isn't optimal, in hindsight. Changing this would need some migration logic. 

- I'm going to take a step back and think about this a bit more. I'm working on the workflow library now. It may be the best time to make some improvements like this.